tests/main.nf.test

```diff
@@ -564,5 +564,64 @@ nextflow_workflow {
             }
         },
 
+        test("cycle: no illumination correction, backsub") {
+
+            when {
+                params {
+                    segmentation = "mesmer"
+                    backsub = true
+                }
+                workflow {
+                    """
+                    input[0] = Channel.of(
+                        [
+                            [id:"TEST1", cycle_number:1, channel_count:4],
+                            "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/imaging/background_subtraction/cycif_tonsil_registered.ome.tif",
+                            [],
+                            [],
+                        ],
+                        [
+                            [id:"TEST2", cycle_number:2, channel_count:4],
```

Comment:
The `id` needs to be `TEST1` here. Otherwise you're declaring two 1-cycle datasets rather than a single 2-cycle dataset as intended.

This points to some problems the marker sheet validation logic -- the marker sheet has more channels than the input data but no error was raised. Looking at the code, I guess once we address "TODO: Validate that cycle_number is 1..N, in order, for all samples." in `PIPELINE_INITIALISATION`, then `validateInputSamplesheetMarkersheet` should catch this issue. Can you add an actual issue so we're not relying just on that TODO?

Reply:
Ah, of course.  Thanks for catching that. I've fixed it in the code.

As discussed, I think validation is ok because I was using the wrong input file for this test.
Please open new comment or issue if I've got this wrong. Thanks!